### PR TITLE
in_tail: Work around "undeletable file" issue on Windows

### PR DESCRIPTION
**Pull request by fujimotos**
Windows will not actually delete a file as long as there is any
open handler for it. Until we close the last open handler, the
file is kept alive as "pending delete".

Obviously this has been breaking the log rotation mechanism for
some users. Work arount that by checking the "DeletePending" flag
manually in tail_fs_check().

Signed-off-by: Fujimoto Seiji <fujimoto@ceptord.net>

### CODE REVIEW

**Comment by farcop** [comment]
Solves https://github.com/fluent/fluent-bit/issues/2133

**Comment by edsiper** [comment]
thanks @fujimotos and @farcop !

File: plugins/in_tail/tail_fs_stat.c [hunk 1]
```diff
@@ -116,6 +116,26 @@ static int tail_fs_check(struct flb_input_instance *ins,
             continue;
         }
 
+#ifdef FLB_SYSTEM_WINDOWS
+        HANDLE h;
+        FILE_STANDARD_INFO info;
+
+        h = _get_osfhandle(file->fd);
```
Comment (sachinmsft) [inline plugins/in_tail/tail_fs_stat.c:123]: should not we close this handle as it is defined as void*

Comment (fujimotos) [reply plugins/in_tail/tail_fs_stat.c:123]: @sachinmsft  According to the documentation:

> To close a file whose operating system (OS) file handle is obtained by _get_osfhandle, call _close on the file descriptor fd. Never call CloseHandle on the return value of this function.
> 
> https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/get-osfhandle?view=vs-2019

So I thought it is not supposed to be closed. Am I reading it wrongly?

Comment (sachinmsft) [reply plugins/in_tail/tail_fs_stat.c:123]: My bad. I did not read the remarks and thought it as a normal file handle. 
no need to close it.

Comment (fujimotos) [reply plugins/in_tail/tail_fs_stat.c:123]: Just OK. Now I resolved this discussion.

**Comment by andrerca** [comment]
Hello,

i'm with the same error when fluent-bit is in different server, reading *.json files from an network share.
File is marked for deletion by log rotate, but it will remain in system if fluent-bit is still running, no matter how old the file is.

**Pull request merged**